### DISCUSSION TITLE
How to manage growing list of IP bans

### DISCUSSION THREAD

**Discussion by pcla56** [fail2ban/fail2ban#3316] [General]
Hi Team, 
I use fail2ban successfully on my LAMP server.
Recently I decided to try and discourage sshd hackers by ban'ing them without release. I have now a large list (approx 3500 and growing) of ban'ed IP addresses when I query the sshd jail.
My concern - hence this discussion - is this must be taking increasing memory for the agent, so I would like to try and offload this increasing list from fail2ban. But I don't see any obvious ways. Presumably this is a familiar problem to some and I would welcome any suggestions please if I can be a bit smarter with defeating the bad guys!
Thx Paul

**Comment by sebres** [comment]
Persistent banning is not recommended at all due to many reasons.
Simplest one - a lot of hosts with dynamic IPs may be affected by some botnet, so if they switching the IPs later, remaining persistently banned, they would still unnecessarily bother your net-filter subsystem and fail2ban.

Take a look at https://github.com/fail2ban/fail2ban/discussions/2952#discussioncomment-414693 or https://stackoverflow.com/a/66443690/7161854 for other option (e. g. `bantime.increment`, which allows to prolong ban for already known, recidive, evildoers and ban them faster by repeated attempts).

As for reducing of memory utilization, you may consider the answers cumulated in https://github.com/fail2ban/fail2ban/issues/2900#issuecomment-752781723 (especially `maxmatches` and `dbmaxmatches` options).

**Reply by sebres** [reply]
Oh, forgotten yet... New version of fail2ban will get a subnet-banning facility at some point, so if several IPs belong to the same subnet, they will be combined to 1 entry.
This is part of #927 (and not yet implemented in stock fail2ban versions).

**Reply by cyb3rz3us** [reply]
> if several IPs belong to the same subnet, they will be combined to 1 entry

This will be a great feature add and personally save me a fair amount of time.

> Persistent banning is not recommended

I would personally disagree with this but I also think it comes down to a case-by-case basis. I've used F2B for well over 3 years and not culled a single entry from my ban list. Again, I think this decision is heavily dependent an how F2B is used but there are definitely use cases where persistent banning is not a bad idea.
